src/app/core/profile/profile.service.ts

```diff
@@ -49,10 +49,25 @@ export class ProfileService {
     }
 
     isLoggedIn(): boolean {
-        return Boolean(this.user && this.user.role);
+        return Boolean(this.user);
     }
 
     isGuest(): boolean {
         return !this.isLoggedIn();
     }
+
+    getUserRole(): string {
+        return this.user.role;
+    }
+
+    hasUserRole(roles: Array<string>): boolean {
+        if (roles && roles.length) {
```

Comment:
Maybe ```Array.isArray(roles)``` would be better here. Your solution still would work for strings :). And even will return proper results, but I am not sure you want it to.